beanstalk_worker/worker_start_scan.py

```diff
@@ -26,212 +26,424 @@
-
-    for i in list_logs:
-        if i == "" or i.startswith("Repodata"):
-            continue
-        i_split = i.split(" ", 1)
-        package_list.append(i_split[0])
-
-    return package_list
-
-
-def scan_job_data(job_data):
-    msg = ""
-    logs = ""
-    json_data = None
-    logger.log(level=logging.INFO, msg="Received job data from tube")
-    logger.log(level=logging.INFO, msg="Job data: %s" % job_data)
-
-    # if job_data.get("tag") != None:
-    #     image_full_name = job_data.get("name") + ":" + \
-    #         job_data.get("image_tag")
-    # else:
-    #     image_full_name = job_data.get("name")
-
-    # Receive and send `name_space` key-value as is
-    namespace = job_data.get('name_space')
-    notify_email = job_data.get('notify_email')
-
-    image_full_name = job_data.get('name')
-    #.split(":")[0] + ":" + \
-    #    job_data.get("tag")
+class ScannerRunner(object):
+    """
+    Scanner runner class handling basic operation and orchestration of
+    multiple scanner handlers
+    """
 
-    logger.log(level=logging.INFO, msg="Pulling image %s" % image_full_name)
-    pull_data = conn.pull(
-        repository=image_full_name
-    )
+    def __init__(self, job_info):
+        self.job_info = job_info
+        self.atomic_object = Atomic()
+        self.pipeline_scanner_object = PipelineScanner()
+        self.scanner_rpmverify_object = ScannerRPMVerify()
 
-    if 'error' in pull_data:
-        logger.log(level=logging.FATAL, msg="Couldn't pull requested image")
-        logger.log(level=logging.FATAL, msg=pull_data)
-        return
+        # Receive and send `name_space` key-value as is
+        self.job_namespace = job_info.get('name_space')
 
-    # logger.log(level=logging.INFO,
-    #            msg="Creating container for image %s" % image_full_name)
+    def pull_image_under_test(self, image_under_test):
+        """
+        Pulls image under test on test run host
+        """
+        logger.log(
+                level=logging.INFO,
+                msg="Pulling image %s" % image_under_test)
+        pull_data = conn.pull(
+            repository=image_under_test
+        )
 
-    # container = conn.create_container(image=image_full_name,
-    #                                      command="yum -q check-update")
+        if 'error' in pull_data:
+            logger.log(level=logging.FATAL, msg="Couldn't pull requested image")
+            logger.log(level=logging.FATAL, msg=pull_data)
+            return False
+        logger.log(level=logging.INFO, msg="Image is pulled %s" % image_under_test)
+        return True
+
+    def run_a_scanner(self, scanner, image_under_test):
+        """
+        Run the given scanner on image_under_test
+        """
+        json_data = {}
+
+        # If scanner is pipeline-scanner then hand over to specific class,
+        # since pipeline-scanner mounts the image first and then executes
+        if scanner == "registry.centos.org/pipeline-images/pipeline-scanner":
```

Comment:
This is unnecessary repetition of code across the scanners. You could do something like:

``` python
class ScannerRunner:
    # OrderedDict if order is important, else just a dict
    scanners = OrderedDict(
        {
            "registry.centos.org/pipeline-images/pipeline-scanner": PipelineScanner,
            "registry.centos.org/pipeline-images/scanner-rpm-verify": ScannerRpmVerify
         }
    )

    def run_scanner(self, name, image_under_test):
        scanner = self.scanners.get(name)()
        # then you could have the following code

    def run_scanners(self, image_under_tests):
        for name, mod in self.scanner.items():
            scanner = mod()
            # run scanners
```